### PR TITLE
Events

### PR DESCRIPTION
**Pull request by PlainOlSoapBar**
- Events can either be public or private.
- Event hosts can be assigned to any person or organization, regardless if they exist in BroncoBond. 
  - By default, the event host will be set as the event creator (username).
# Public Events
- Can be created only by administrators.
- Can be edited or deleted only by the event creator.
- Can be searched for by users.
- Users can express interest in these events.
# Private Events
- Can be created by anyone.
- Can be edited and deleted only by the event creator. 
- Administrators can delete any private event.

### CODE REVIEW

**Comment by monsothin96** [comment]
Jay-Branch

File: backends/controller/event.controller.js
Comment (Jayvnn) [inline backends/controller/event.controller.js:1]: When updating event, check if the new event start date time is after the current date time.

Comment (Jayvnn) [inline backends/controller/event.controller.js:1]: Also during the updating event, check if the new start date time and new end date time are appropriate before creating the model.

Comment (Jayvnn) [inline backends/controller/event.controller.js:1]: When deleting event first check the type of event, then organize code by if event is private or public

**Review by Jayvnn — CHANGES_REQUESTED** [review]
Check comments on needed changes

**Review by Jayvnn — APPROVED** [review]

**Pull request merged**